pkg/telemetry/workflowsk8s.go

```diff
@@ -153,10 +156,16 @@ func NewClusterStateWorkflow(d dynamic.Interface, rm meta.RESTMapper) (Workflow,
 	for _, oo := range optionalObjects {
 		provider, err := oo.providerCreator(string(oo.countKey), d, rm)
 		if err != nil {
-			if !meta.IsNoMatchError(err) {
-				return nil, err
-			}
 			// If there's no kind for object in the cluster then just don't add its count provider.
+			if meta.IsNoMatchError(err) {
+				continue
+			}
+			if errG := (&discovery.ErrGroupDiscoveryFailed{}); errors.As(err, &errG) {
+				continue
+			}
```

Comment:
Nit: I'd consider moving this logic to `NewK8sObjectCountProviderWithRESTMapper` as it's closer to the error returned from the `RESTMapper`. If any of the `{ErrGroupDiscoveryFailed, NoMatchErr, ...}` is returned there, we could return either a boolean indicating the existence of this kind or define our own umbrella error type, e.g. `NoKindForObjectInCluster` and return it.

Reply:
After some thought, I've changed this to return a specific error just as you suggested, and only that error is being checked now in here. PTAL